x-pack/plugins/observability_solution/apm/public/components/app/onboarding/instructions_set.tsx

```diff
@@ -41,9 +42,18 @@ export function InstructionsSet({
 }) {
   const tabs = getTabs(instructions.instructionVariants);
 
-  const [selectedTab, setSelectedTab] = useState<string>(tabs[0].id);
+  const { hash } = useLocation();
```

Comment:
I think you should probably use `useApmParams`.

Reply:
I knew APM probably handled all of this kind of use case already. I'll redress.

Reply:
Check out https://github.com/elastic/kibana/pull/180843/commits/1ec9265b42070afb074664b49c3e7f03565eecae and let me know if this looks better now.